python/_tskitmodule.c

```diff
@@ -9646,6 +9646,80 @@ TreeSequence_k_way_stat_method(TreeSequence *self, PyObject *args, PyObject *kwd
     return ret;
 }
 
+static PyObject *
+TreeSequence_weighted_stat_vector_method(
+    TreeSequence *self, PyObject *args, PyObject *kwds, weighted_vector_method *method)
+{
+    PyObject *ret = NULL;
+    static char *kwlist[]
+        = { "weights", "windows", "mode", "span_normalise", "centre", NULL };
+    PyObject *weights = NULL;
+    PyObject *windows = NULL;
+    PyArrayObject *weights_array = NULL;
+    PyArrayObject *windows_array = NULL;
+    PyArrayObject *result_array = NULL;
+    tsk_size_t num_windows;
+    npy_intp *w_shape, result_shape[3];
+    tsk_flags_t options = 0;
+    tsk_size_t num_samples;
+    char *mode = NULL;
+    int span_normalise = true;
+    int centre = true;
+    int err;
+
+    if (TreeSequence_check_state(self) != 0) {
+        goto out;
+    }
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|sii", kwlist, &weights, &windows,
+            &mode, &span_normalise, &centre)) {
+        goto out;
```

Comment:
Looks like we're not doing bad parameter testing in test_lowlevel.py. Doesn't have to be comprehensive, but we do need to make sure the error path is covered (it's a common and hard to track down source of segfaults)

Reply:
Ah - I got confused about that.